### PR TITLE
add doxygen documentation

### PR DESCRIPTION
**Pull request by mayank-sirotiya-imgtec**
Signed-off-by: Mayank Sirotiya <Mayank.Sirotiya@imgtec.com>

### CODE REVIEW

File: docs/doxyfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+PROJECT_NAME           = "device_manager"
+PROJECT_NUMBER         = 0.9.0
+JAVADOC_AUTOBRIEF      = YES
+HTML_HEADER            = /home/mayank/mak500/work/github_repos/openwrt_verify_spi/packages/device-manager/docs/creator_header.html
```
Comment (abhijit-mahajani-imgtec) [inline docs/doxyfile]: Is this correct?
/home/mayank/mak500 ?

File: docs/mainpage.dox [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/** @mainpage Flow Device Manager
```
Comment (abhijit-mahajani-imgtec) [inline docs/mainpage.dox]: we decided not to put mainpage.dox, since readme.md is giving the overview

**Comment by rahul-daga-imgtec** [comment]
Approved + Verified

**Comment by abhijit-mahajani-imgtec** [comment]
Approved and Verified

**Pull request merged**